### ISSUE TITLE
Provide smcp ß by character substitution instead of a ligature

### ISSUE THREAD

**Issue by katef** [skosch/Crimson#36]
I think we have almost exactly the opposite situation to #34 for ß - that is, a ligature which I think ought to be substituted for two characters.

Currently `germandbls.sc` (can we name it `esset` instead?) is drawn as two capital 'S' characters, which I think is correct because ß is lowercase, and so would become two S characters on converting to capitals.

However I would rather provide `smcp` for ß as a substitution to two `s.sc` glyphs instead, and remove `germandbls.sc`.

The reason is that I'd rather avoid having pictures of two characters with a space between them because that space would be incorrect if a user adjusts the letter spacing. That's all the more likely for small caps, since at larger sizes spacing between capitals is often made a little looser.

I'd implement this after #29.

**Comment by katef** [comment]
Er, eszett, that is. My spelling isn't great in English, let alone German :)

Alternately, I think this style would suit Crimson:
![sz](http://typography.guru.bucket01.s3.amazonaws.com/monthly_2015_01/kloesse.thumb.jpg.9f4899bba405c0588d964fadd9d2ad29.jpg)

With a nice rundown here: http://typography.guru/journal/how-to-draw-a-capital-sharp-s-r18/

**Comment by skosch** [comment]
Yeah, you have a point in that users can't track out the SS. I'm happy with your suggestion, if you can make it work. I've never seriously looked into how OpenType features work, save for simple ligature substitutions so I didn't even realize what you suggested was an option.

I find that capital ß charming. Growing up in Germany I very very rarely encountered it – for some inexplicable reason I associate it with mid-century Eastern Germany – but it's growing on me the more I look at it. Feel free to give it a shot, commit it and I'll fiddle around with it until we're both happy with it?

**Comment by katef** [comment]
It would also be possible to implement `smcp` to substitute ß for two `s.sc` characters, with a capital ß glyph instead as a stylistic alt. If the capital glyph feels so charming as to be old fashioned, maybe that's a better approach.

Personally I really like the glyph! The angle reminds me of the middle part of "M". I will try drawing it, but not right now.

I'm working on getting #29 into shape, by the way. Spotting these things is a side effect of that.

**Comment by katef** [comment]
Hello from Åland! Happy new year!

I had a go at drawing out the above idea for capital ß in Crimson's serif style. Please note this is to smallcaps proportion; it is not uppercase. So of course it looks relatively quite wide.

I don't think it's all quite there yet (particularly the joint top-right is a bit awkward), but I'm thinking of it calligraphically, in terms of how I might angle an oblique pen nib, and I think the idea is potentially useful. I did find it hard to make the angled stroke elicit the feel of the `z.sc`, which I think it ought to.

![germandbls sc1](https://cloud.githubusercontent.com/assets/1371085/12080925/bfb52f28-b262-11e5-8a64-39addf3e2f24.png)

**Comment by skosch** [comment]
Happy new year! Love it so far, great start. I would …

* even out the inner left curve a little bit (the left handle of the horizontal bezier extends past the vertical point – I try to avoid that, because it creates an extra inflection in the curve)
* flare out the curved part (the incoming stroke) going into the top right transition, to help emphasize the calligraphic thick-thin constrast. In particular, I would use this opportunity to align that corner with the bowl more (i.e. pull it to the right, or tuck the bowl closer to the stem)
* commit to sharp corners at the top right transition, i.e. angle the bezier handles into the curve instead of pointing them straight up
* same for the inward-pointing transition below. Just use sharp corner points and play with the handles (that'll also help with the wobbliness of the curve in that segment)
* the bowl (belly?) can definitely be heavier; and the final serif (tail?) maybe a tiny bit smaller … or maybe not. You're trying to optically match the weight of the vertical stem.

Overall I think the skeleton is lovely, it just needs some cosmetic attention. My heuristic to avoid clumsy-looking curves is to keep the two opposing bezier handles defining a segment more or less balanced (and always on the same side of the curve), and un-balance them only if there's a good reason to.

Hope that helps! :)
